Don't change ld behaviour by adding -brtl flag here.
Work around ld bug with -bsvr4 -bnoentry -bE:file args.
--- ld/aixplugin.c.orig	2015-02-10 17:28:17 +0100
+++ ld/aixplugin.c	2015-02-10 17:33:55 +0100
@@ -32,7 +32,7 @@
 	String const *argString;
 	String *newString = NULL;
 	char libpath2runpath = 1; /* whether to record -L as runpath */
-	char svr4mode = 0; /* whether we are in -bsvr4 mode */
+	String const *svr4modeArg = NULL; /* whether we are in -bsvr4 mode */
 	char const *argBuffer;
 	int argBufferLength;
 
@@ -53,20 +53,22 @@
 		 || StringListAppendConcat(aixRunpathList, "/lib", strlen("/lib"), NULL) < 0
 		) break;
 
-		/* always use runtime linking (-brtl) */
-		if (StringListAppendConcat(data->out->argList, "-brtl", strlen("-brtl"), NULL) < 0)
-			break;
-
 		/* detect if we are in svr4 mode (-bsvr4).
 		 * "-R runpath" is only accepted in svr4 mode,
 		 * but we add "-R runpath" as extension in normal mode to add default runpath.
+		 * Also, AIX ld has this problem:
+		 *   $ /usr/ccs/bin/ld -bsvr4 -bE:xx.exp -bnoentry xx.o
+		 *   ld: 0706-005 Cannot find or open file: l
+		 *       ld:open(): No such file or directory
+		 * Simplest workaround is to put -bsvr4 last.
 		 */
 		for(argc = 1; argc < StringListGetSize(data->in->argList); argc++) {
 			argString = StringListGetString(data->in->argList, argc);
 			argBuffer = StringGetBuffer(argString);
 			if (strcmp(argBuffer, "-bsvr4") == 0) {
-				svr4mode = 1;
+				svr4modeArg = argString;
 				libpath2runpath = 0;
+				break;
 			}
 		}
 
@@ -80,7 +82,7 @@
 				libpath2runpath = 0;
 				/* This also removes each runpath specified before. */
 				StringListClear(runpathList);
-				if (svr4mode) {
+				if (svr4modeArg) {
 					StringListClear(defaultRunpathList);
 				} /* else we accept -R as extension */
 				continue;
@@ -96,7 +98,7 @@
 				/* '-blibpath:' kills previous '-blibpath:' */
 				StringListClear(runpathList);
 
-				if (svr4mode) {
+				if (svr4modeArg) {
 					/* it also kills previous '-Rrunpathlist' in svr4mode */
 					StringListClear(defaultRunpathList);
 				} /* else we accept -R as extension */
@@ -125,7 +127,7 @@
 				continue;
 			} else
 			if (strncmp(argBuffer, "-R", strlen("-R")) == 0) {
-				if (svr4mode) {
+				if (svr4modeArg) {
 					/* in svr4 mode, '-R' kills previous '-blibpath:' arguments.
 					 * in normal mode, we implement '-R runpath' as an extension.
 					 */
@@ -150,6 +152,10 @@
 					break;
 
 				continue;
+			} else
+			if (strcmp(argBuffer, "-bsvr4") == 0) {
+				/* add -bsvr4 last */
+				continue;
 			}
 
 			/* keep other arguments */
@@ -193,6 +199,9 @@
 		if (StringListAppendListModify(data->out->argList, data->in->sysLibpath, 0, -1, "-L", 2, NULL, 0) < 0)
 			break;
 
+		if (svr4modeArg && StringListAppendString(data->out->argList, svr4modeArg) < 0)
+			break;
+
 		err = 0;
 	} while(0);	/* end dummy loop */
 
